### PR TITLE
GET /api/articles/:article_id/comments endpoint

### PR DESCRIPTION
**Pull request by EmoryBrock**
this pull request generates the following:

- creates a GET /api/articles/:article_id/comments endpoint
- modified associated articles controller and model JS files with appropriate functions and callbacks
- utilises TDD tests for:
        - returns 200 status & an array of comment objects with requested properties for specified article id
        - error test 404 if provided an article_id that doesn't exist in database.
        - error test 400 if provided an article_id that is invalid id.

### CODE REVIEW

File: controllers/articles.controller.js [hunk 1]
```diff
@@ -19,4 +21,20 @@ exports.getArticles = (req, res, next) => {
...
     })
+}
+
+exports.getCommentsByArticleID = (req, res, next) => {
+    const {article_id }= req.params
+
+    if (isNaN(article_id)) return next ({
+        status: 400,
+        message: `bad request: this is not a number`
+    })
```
Comment (aloush-dev) [inline controllers/articles.controller.js]: PSQL would throw an error here

File: models/articles.model.js [hunk 2]
```diff
@@ -34,4 +34,23 @@ exports.fetchArticles = () => {
...
+        .query(
+            `SELECT * FROM comments
+            WHERE article_id=$1;`, [id]) 
+        .then(({rows}) => {
+                if (rows.length === 0) {
+                    return Promise.reject({ 
+                        status: 404, 
+                        message: `No comments found. Article ID ${id} does not exist.`
+                    })
+                }
```
Comment (aloush-dev) [inline models/articles.model.js]: If an empty array comes back its not an error it just means that there is no comments with that article which you would want to send back to the client.

**Review by aloush-dev — CHANGES_REQUESTED** [review]
I'd like to see a 200 test.
If an article has no comments ideally you want to send an empty array back.

Before you send your request off to get the comments you'd want to know that the article exists first.
You have a model already that checks for an article by ID

File: controllers/articles.controller.js [hunk 3]
```diff
@@ -19,4 +21,16 @@ exports.getArticles = (req, res, next) => {
...
     })
+}
+
+exports.getCommentsByArticleID = (req, res, next) => {
+    const {article_id }= req.params
+
+    Promise.all([fetchArticleById(article_id), fetchCommentsByArticleId(article_id)])
+        .then((results) => {
+            res.status(200).send(results[1])
+            return Promise.all([fetchArticleById(article_id), fetchCommentsByArticleId(article_id)])
```
Comment (yalicee) [inline controllers/articles.controller.js:32]: What is this line doing?

**Review by yalicee — APPROVED** [review]
could you explain to what the promise.all is doing on line 32? Because you seem to have already done it on line 29. However I can see that this works so approved anyway

File: controllers/articles.controller.js (hunk 3 above)
Comment (EmoryBrock) [reply controllers/articles.controller.js:32]: Summing our direct messages here.

It is to verify if the article_id exists in the database and if so, go on to fetch the comments.  As the promiseAll returns 2 results in the array, a the successful fetch of comments is in the 1 index of the array.  Also, error if the first promise (fails to find article with the id), the error is then passed along and handled appropriately.

You suggested using a promise chain to achieve the same result.

**Pull request merged**